packages/reason-expo/src/Location.re

```diff
@@ -60,37 +62,39 @@ type coords = {
   speed: float,
 };
 
-[@bs.deriving abstract]
 type location = {
   coords,
   timestamp: int,
 };
 
-[@bs.deriving abstract]
-type getCurrentPositionAsyncOptions = {
-  accuracy: Accuracy.t,
-  maximumAge: int,
-};
+type getCurrentPositionAsyncOptions;
+[@bs.obj]
+external getCurrentPositionAsyncOptions:
+  (~maximumAge: int=?, ~accuracy: Accuracy.t=?) =>
+  getCurrentPositionAsyncOptions;
```

Comment:
I'm not super familiar with this, could you give me 2 cents?

Reply:
This works the same as `@react.component` bindins. It creates a function to construct record and gives you ability to use cool tricks like omitting optional fields. It's similar to `@bs.deriving abstract`
You can use it as:
```reasonml
getCurrentPositionAsyncOptions()
getCurrentPositionAsyncOptions(~maximumAge=1)
getCurrentPositionAsyncOptions(~maximumAge=1, ~accuracy=Accuracy.Low)
```
and it always creates correct record.

Reply:
Thanks! It's cool